src/assessments/seo/keyphraseLengthAssessment.js

```diff
@@ -1,31 +1,99 @@
-	var keyphraseLength = researcher.getResearch( "keyphraseLength" );
-
-	var assessmentResult = new AssessmentResult();
-
-	if ( ! paper.hasKeyword() ) {
-		assessmentResult.setScore( -999 );
-		assessmentResult.setText( i18n.dgettext( "js-text-analysis", "No focus keyword was set for this page. " +
-			"If you do not set a focus keyword, no score can be calculated." ) );
-	} else if ( keyphraseLength > 10 ) {
-		assessmentResult.setScore( 0 );
-		assessmentResult.setText( i18n.dgettext( "js-text-analysis", "The keyphrase is over 10 words, a keyphrase should be shorter." ) );
+ * Assessment to check whether the keyphrase has a good length.
+ */
+class KeyphraseLengthAssessment extends Assessment {
+	/**
+	 * Sets the identifier and the config.
+	 *
+	 * @param {Object} config The configuration to use.
+	 *
+	 * @returns {void}
+	 */
+	constructor( config = {} ) {
+		super();
+
+		let defaultConfig = {
+			scores: {
+				keyphraseLength: 0,
+			},
+			assessment: {
+				recommendedMaximum: 5,
+				acceptableMaximum: 9,
+			},
+		};
+
+		this.identifier = "keyphraseLength";
+		this._config = merge( defaultConfig, config );
+	}
+
+	/**
+	 * Assesses the keyphrase presence and length.
+	 *
+	 * @param {Paper} paper The paper to use for the assessment.
+	 * @param {Researcher} researcher The researcher used for calling research.
+	 * @param {Jed} i18n The object used for translations.
+	 * @returns {AssessmentResult} The result of this assessment.
+	 */
+	getResult( paper, researcher, i18n ) {
+		this._keyphraseLength = researcher.getResearch( "keyphraseLength" );
+		let assessmentResult =  new AssessmentResult();
+
+		if ( paper.hasKeyword() ) {
+			assessmentResult.setScore( this.calculateScore() );
+			assessmentResult.setText( this.translateScore( i18n ) );
+		} else  {
+			assessmentResult.setScore( -999 );
+			assessmentResult.setText( i18n.dgettext( "js-text-analysis", "No focus keyword was set for this page. " +
+				"If you do not set a focus keyword, no score can be calculated." ) );
+		}
+		return assessmentResult;
+	}
+
+	/**
+	 * Calculates the score result based on the keyphraseLength research.
+	 * @returns {Object} object with score and text.
+	 */
+	calculateScore() {
+
+		if ( inRange( this._keyphraseLength, 1, this._config.assessment.recommendedMaximum ) ) {
+			return 9;
+		}
+
+		if ( inRange( this._keyphraseLength, this._config.assessment.recommendedMaximum, this._config.assessment.acceptableMaximum ) ) {
+			return 6;
+		}
+
+		if ( this._keyphraseLength >= this._config.assessment.acceptableMaximum ) {
+			return 3;
+		}
 	}
 
-	return assessmentResult;
+	/**
+	 * Translates the score result based into specific feedback to the user.
+	 * @param {object} i18n The i18n-object used for parsing translations.
+	 * @returns {Object} object with score and text.
+	 */
+	translateScore( i18n ) {
+		if ( inRange( this._keyphraseLength, 1, this._config.assessment.recommendedMaximum ) ) {
```

Comment:
For reasons of consistency, I'd tend to also add the `1` as a  `recommendedMinimum`